app/rest/model/src/main/java/io/syndesis/model/integration/IntegrationRevision.java

```diff
@@ -16,61 +16,121 @@
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
-import org.immutables.value.Value;
+import io.syndesis.core.SuppressFBWarnings;
+import io.syndesis.model.Kind;
+import io.syndesis.model.WithId;
+import io.syndesis.model.WithKind;
+import io.syndesis.model.WithName;
 
 import java.math.BigInteger;
+import java.util.Collections;
 import java.util.Date;
+import java.util.List;
 import java.util.Optional;
 
-@Value.Immutable
 @JsonDeserialize(builder = IntegrationRevision.Builder.class)
-public abstract class IntegrationRevision {
+public class IntegrationRevision implements WithKind, WithId<IntegrationRevision>, WithName {
 
-    /**
-     * The revision number. This is unique per {@link Integration}.
-     * Once an {@IntegrationRevision} gets a version, it should not be mutated anymore.
-     * @return
-     */
-    public abstract Optional<Integer> getVersion();
+    private final Optional<String> id;
+    private final Optional<Integer> version;
+    private final Optional<String> integrationId;
 
-    /**
-     * The version of the integration revision which was the origin of this revision.
-     * @return 0 if this is the first revision of an integration, the parent version otherwise.
-     */
-    public abstract Integer getParentVersion();
+    private final String name;
+    private final Integer parentVersion;
 
-    public abstract IntegrationRevisionSpec getSpec();
+    private final IntegrationRevisionSpec spec;
+    private final List<String> stepsDone;
 
-    /**
-     * The desired state of the revision.
-     * @return
-     */
-    public abstract IntegrationRevisionState getTargetState();
+    private final IntegrationRevisionState currentState;
+    private final IntegrationRevisionState targetState;
 
-    /**
-     * The current state of the revision.
-     * @return
-     */
-     public abstract IntegrationRevisionState getCurrentState();
+    private final BigInteger timesUsed;
+    private final Date lastUpdated;
+    private final Date createdDate;
 
-    /**
-     * Message describing the currentState further (e.g. error message)
-     * @return
-     */
-    public abstract Optional<String> getCurrentMessage();
+    public IntegrationRevision() {
+        this(Optional.empty(), Optional.of(1), null, null, null, Collections.emptyList(), null, null, null, null, null);
+    }
 
-    /**
-     * Message which should become the currentMessage after reconciliation
-     * @return
-     */
-    public abstract Optional<String> getTargetMessage();
+    public IntegrationRevision(Optional<String> integrationId, Optional<Integer> version, String name, Integer parentVersion, IntegrationRevisionSpec spec, List<String> stepsDone, IntegrationRevisionState currentState, IntegrationRevisionState targetState, BigInteger timesUsed, Date lastUpdated, Date createdDate) {
+        this.id = integrationId.map(i -> i + "#" + version.orElse(1));
```

Comment:
using # is kinda bad, you have to encode # from the UI since # is a special query token for browsers.  Perhaps ":" would be better.